frappe/modules/full_text_search.py

```diff
@@ -10,97 +10,219 @@
 from bs4 import BeautifulSoup
 from frappe.website.render import render_page
 from frappe.utils import set_request, cint
-from frappe.utils.global_search import get_routes_to_index
+import os
+
+class FullTextSearch:
+	""" Frappe Wrapper for Whoosh """
+
+	def __init__(self, index_name):
+		self.index_name = index_name
+		self.index_path = get_index_path(index_name)
+		self.schema = Schema(
+			title=TEXT(stored=True), path=ID(stored=True), content=TEXT(stored=True)
+		)
+
+	def get_routes_to_index(self):
+		"""Get all routes to be indexed, this includes the static pages
+		in www/ and routes from published documents
+
+		Returns:
+			self (object): FullTextSearch Instance
+		"""
+		routes = get_static_pages_from_all_apps()
+		routes += get_doctype_routes_with_web_view()
+		return routes
+
+	def build(self):
+		"""	Build search index for all web routes """
+
+		print("Building search index for all web routes...")
+		routes = self.get_routes_to_index()
+		self.documents = [self.get_document_to_index(route) for route in routes]
+		self.build_index()
+
+	def get_document_to_index(self, route):
+		"""Render a page and parse it using BeautifulSoup
+
+		Args:
+			path (str): route of the page to be parsed 
+
+		Returns:
+			document (_dict): A dictionary with title, path and content
+		"""
+		frappe.set_user("Guest")
+		frappe.local.no_cache = True
+
+		try:
+			set_request(method="GET", path=route)
+			content = render_page(route)
+			soup = BeautifulSoup(content, "html.parser")
+			page_content = soup.find(class_="page_content")
+			text_content = page_content.text if page_content else ""
+			title = soup.title.text.strip() if soup.title else route
+
+			frappe.set_user("Administrator")
```

Comment:
why is this being set? even the one to set user as Guest on line 53 seems unnecessary. is there a reason for this being set?

Reply:
Some info on website is generated only for specific users, and Admin can see most of those. So we set the user to guest to ensure we're never indexing any such data.

The changing back to Admin is also important, since there are other processes post migrate that need admin privileges. This could be handled in other places, but it's just simpler to do so in the function itself. The cost of this is minimal, and either way this is always pushed to background or is run in migrate

Reply:
Just make sure that there's no way to access this function as a user running the application. There is a possibility of privilege escalations on the running thread if that happens.